vmdk_plugin/drivers/vmdk/vmdk_driver.go

```diff
@@ -354,56 +373,40 @@ func (d *VolumeDriver) Create(r volume.Request) volume.Response {
 	log.WithFields(log.Fields{"name": r.Name,
 		"fstype": r.Options["fstype"]}).Info("Attaching volume and creating filesystem ")
 
-	watcher, skipInotify := fs.DevAttachWaitPrep(r.Name, watchPath)
+	watchCtx, errWatch := fs.DevAttachWaitPrep()
+	if errWatch != nil {
+		log.WithFields(log.Fields{"Name": r.Name, "error": errWatch}).Warning("Failed to watch disk attach ")
+	}
 
-	dev, errAttach := d.ops.Attach(r.Name, nil)
+	volDev, errAttach := d.ops.Attach(r.Name, nil)
 	if errAttach != nil {
 		log.WithFields(log.Fields{"name": r.Name,
 			"error": errAttach}).Error("Attach volume failed, removing the volume ")
 		// An internal error for the attach may have the volume attached to this client,
 		// detach before removing below.
-		d.ops.Detach(r.Name, nil)
-		errRemove := d.ops.Remove(r.Name, nil)
-		if errRemove != nil {
-			log.WithFields(log.Fields{"name": r.Name, "error": errRemove}).Warning("Remove volume failed ")
-		}
+		d.forceDetachAndRemove(r.Name, nil)
```

Comment:
detachAndRemove()? - the function is doing that and there is no force option. The second arg is nil always in all places where this function is called and there are really no options to pass for a detach and/or remove request to the ESX service.

Second arg. is not needed and should be removed.